src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java

```diff
@@ -235,41 +236,66 @@ public Pipeline createPipelineAggregated(ItemGroup context, boolean showAggregat
      *
      * @param noOfPipelines number of pipeline instances
      */
-    public List<Pipeline> createPipelineLatest(int noOfPipelines, ItemGroup context, boolean pagingEnabled) {
+    public List<Pipeline> createPipelineLatest(int noOfPipelines, ItemGroup context, boolean pagingEnabled,
+                                               boolean showUpstream) throws PipelineException {
         List<Pipeline> result = new ArrayList<Pipeline>();
-        int no = noOfPipelines;
-        if (firstProject.isInQueue()) {
-            String pipeLineTimestamp = PipelineUtils.formatTimestamp(firstProject.getQueueItem().getInQueueSince());
-            List<Stage> pipelineStages = new ArrayList<Stage>();
-            for (Stage stage : getStages()) {
-                pipelineStages.add(stage.createLatestStage(context, null));
+        if (showUpstream) {
```

Comment:
Could we refactor this block and extract to its own method? Same goes for the else block.